src/files/BrsFile.ts

```diff
@@ -1064,6 +1260,93 @@ export class BrsFile {
         }
     }
 
+    public getSignatureHelp(statement: FunctionStatement | ClassMethodStatement) {
+        const func = statement.func;
+        const funcStartPosition = func.range.start;
+
+        // Get function comments in reverse order
+        let currentToken = this.getTokenAt(funcStartPosition);
+        let functionComments = [] as string[];
+        while (true) {
+            currentToken = this.getPreviousToken(currentToken);
+            if (!currentToken) {
+                break;
+            }
+            if (currentToken.range.start.line + 1 < funcStartPosition.line) {
+                if (functionComments.length === 0) {
+                    break;
+                }
+            }
+
+            const kind = currentToken.kind;
+            if (kind === TokenKind.Comment) {
+                // Strip off common leading characters to make it easier to read
+                const commentText = currentToken.text.replace(/^[' *\/]+/, '');
+                functionComments.unshift(commentText);
+            } else if (kind === TokenKind.Newline) {
+                if (functionComments.length === 0) {
+                    continue;
+                }
+                // if we already had a new line as the last token then exit out
+                if (functionComments[0] === currentToken.text) {
+                    break;
+                }
+                functionComments.unshift(currentToken.text);
+            } else {
+                break;
+            }
+        }
+        const documentation = functionComments.join('').trim();
+
+        const lines = util.splitStringIntoLines(this.fileContents);
+
+        const params = [] as ParameterInformation[];
+        for (const param of func.parameters) {
+            params.push(ParameterInformation.create(param.name.text));
+        }
+
+        const label = util.getTextForRange(lines, util.createRangeFromPositions(func.functionType.range.start, func.body.range.start)).trim();
+        const signature = SignatureInformation.create(label, documentation, ...params);
+        return signature;
+    }
+
+    public async getReferences(position: Position) {
+        await this.isReady();
+
+        const callSiteToken = this.getTokenAt(position);
+
+        let locations = [] as Location[];
+
+        // No need to actually look if they didn't select a token we can search against
+        if (callSiteToken?.kind !== TokenKind.Identifier) {
```

Comment:
Same comment as [this one](https://github.com/rokucommunity/brighterscript/pull/191/files#r520495177)...you need to check for more than just `TokenKind.Identifier`